build_tools/cmake/setup_sccache.sh

```diff
@@ -0,0 +1,55 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# ccache (https://ccache.dev/) setup, with read/write + local/remote options.
+#
+# Defaults to only reading from the shared remote cache (hosted on GCS) used by
+# our Linux CI. The postsubmit CI writes to the cache, for presubmit CI and
+# local builds to read from.
+#
+# Local caches can also be used to interface with external remote caches
+# (like https://github.com/actions/cache) by
+#   1. downloading the cache directory
+#   2. sourcing with IREE_READ_LOCAL_SCCACHE=1 IREE_WRITE_LOCAL_SCCACHE=[0,1]
+#   3. building with CMake
+#   4. uploading the cache directory (if writing)
+#
+# Note: this file must be *sourced* not executed.
+
+set -euo pipefail
+
+# Configuration environment variables.
+IREE_READ_REMOTE_SCCACHE="${IREE_READ_REMOTE_SCCACHE:-1}"
+IREE_WRITE_REMOTE_SCCACHE="${IREE_WRITE_REMOTE_SCCACHE:-0}"
+IREE_READ_LOCAL_SCCACHE="${IREE_READ_LOCAL_SCCACHE:-0}"
+IREE_WRITE_LOCAL_SCCACHE="${IREE_WRITE_LOCAL_SCCACHE:-0}"
+
+if (( ${IREE_WRITE_REMOTE_SCCACHE} == 1 && ${IREE_READ_REMOTE_SCCACHE} != 1 )); then
+  echo "Can't have 'IREE_WRITE_REMOTE_SCCACHE' (${IREE_WRITE_REMOTE_SCCACHE})" \
+       " set without 'IREE_READ_REMOTE_SCCACHE' (${IREE_READ_REMOTE_SCCACHE})"
+fi
+if (( ${IREE_WRITE_LOCAL_SCCACHE} == 1 && ${IREE_READ_LOCAL_SCCACHE} != 1 )); then
+  echo "Can't have 'IREE_WRITE_LOCAL_SCCACHE' (${IREE_WRITE_LOCAL_SCCACHE})" \
+       " set without 'IREE_READ_LOCAL_SCCACHE' (${IREE_READ_LOCAL_SCCACHE})"
+fi
+
+if (( IREE_READ_REMOTE_SCCACHE == 1 || IREE_READ_LOCAL_SCCACHE == 1 )); then
+  export IREE_USE_SCCACHE=1
+  export CMAKE_C_COMPILER_LAUNCHER="$(which sccache)"
+  export CMAKE_CXX_COMPILER_LAUNCHER="$(which sccache)"
+  sccache --zero-stats
+  sccache --show-stats
+else
+  export IREE_USE_SCCACHE=0
+fi
+
+if (( IREE_READ_LOCAL_SCCACHE == 1 && IREE_WRITE_LOCAL_SCCACHE == 0 )); then
+  export SCCACHE_READONLY=1
+fi
```

Comment:
I don't think this environment variable is used by sccache. See https://github.com/mozilla/sccache/blob/main/docs/Configuration.md for the options it supports.